packages/eslint-plugin-svelte/src/rules/prefer-const.ts

```diff
@@ -0,0 +1,99 @@
+import { createRule } from '../utils';
+import type { TSESTree } from '@typescript-eslint/types';
+
+import { defineWrapperListener, getCoreRule } from '../utils/eslint-core';
+
+const coreRule = getCoreRule('prefer-const');
+
+type Declaration = TSESTree.Expression;
+
+/**
+ * Finds and returns the callee of a declaration node within variable declarations or object patterns.
+ */
+function findDeclarationCallee(node: Declaration) {
+	const { parent } = node;
+	if (parent.type === 'VariableDeclarator' && parent.init?.type === 'CallExpression') {
+		return parent.init.callee;
+	}
+
+	if (
```

Comment:
What code passes through this branch?

Reply:
After investigating for a bit, I realised that there's no valid JavaScript/TypeScript syntax that would create an AST structure, taking into account that the node that's passed into `findDeclarationCallee` comes from a `VariableDeclaration.declarations[number].init`.  The structure doesn't make sense because a `Property` node’s parent cannot simultaneously be a `VariableDeclarator`. 

I'll delete this branch.